src/datagrid/DataGrid.tsx

```diff
@@ -639,22 +651,30 @@ export class DataGrid extends React.PureComponent<DataGridProps, DataGridState>
     };
 
     // Function to handle CustomPageSize change in input box
-    private handleCustomPageSizeChange = () => {
-        const {maxCustomPageSize, totalItems} = this.state.pagination!;
-        const pageSize = this.customPageSizeRef.current && parseInt(this.customPageSizeRef.current.value);
+    private handleCustomPageSizeChange = (event?: any) => {
+        const {maxCustomPageSize = DEFAULT_MAX_PAGE_SIZE, totalItems, pageSize, currentPage} = this.state.pagination!;
+        const customPageSize = this.customPageSizeRef.current && this.customPageSizeRef.current.value;
 
-        if (pageSize && maxCustomPageSize) {
-            if (pageSize <= maxCustomPageSize) {
-                this.getPage(this.state.pagination!.currentPage, pageSize);
-            }
-            //If page size selected by user is greater than maximum limit for custom page size
-            else if (pageSize > maxCustomPageSize && this.customPageSizeRef.current) {
-                this.customPageSizeRef.current.value =
-                    totalItems < maxCustomPageSize ? totalItems.toString() : maxCustomPageSize.toString();
-                this.getPage(DEFAULT_CURRENT_PAGE_NUMBER, parseInt(this.customPageSizeRef.current.value));
+        if (pageSize && customPageSize) {
+            if (!isNumber(customPageSize)) {
+                this.customPageSizeRef.current!.value = pageSize.toString();
+            } else {
+                const customPageSizeInt: number = parseInt(customPageSize);
+                if (pageSize !== customPageSizeInt) {
+                    if (customPageSizeInt <= maxCustomPageSize) {
+                        this.getPage(currentPage, customPageSizeInt);
+                    }
+                    //If page size selected by user is greater than maximum limit for custom page size
+                    else if (customPageSizeInt > maxCustomPageSize && this.customPageSizeRef.current) {
+                        this.customPageSizeRef.current.value =
+                            totalItems < maxCustomPageSize ? totalItems.toString() : maxCustomPageSize.toString();
+                        this.getPage(DEFAULT_CURRENT_PAGE_NUMBER, customPageSizeInt);
+                    }
+                }
```

Comment:
I tried testing this with custom page size as 12345, value in input field doesn't change to 1000 (max limit allowed)

Reply:
Fixed now. Thanks